### PR TITLE
fix(cli): distinguish zero role matches from an unbuilt graph in roles --role

### LINKED ISSUE #2390: bug(cli): roles --role <X> with zero matches says "Run codegraph build first" on a fully-built graph
## Summary

`codegraph roles --role <X>` tells the user to rebuild the graph when the real situation is simply that no symbol has that role. Five of the fourteen agents in the org rollout hit this and several spent time re-running `codegraph build` before concluding the graph was fine.

## Root cause

`src/presentation/queries-cli/overview.ts:354`:

```ts
if (data.count === 0) {
  console.log('No classified symbols found. Run "codegraph build" first.');
  return;
}
```

`data.count === 0` conflates two different states: *the graph has no classified symbols at all* and *this role filter matched nothing*. The advice is actively wrong in the second case — rebuilding changes nothing.

## Reproduction

On any freshly-built graph with no `entry`-classified symbols:

```console
$ codegraph build
Native build orchestrator completed: 943 nodes, 1711 edges, 23 files

$ codegraph roles -T
Node roles (548 symbols):        <- graph is fine

$ codegraph roles --role core -T
… 422 results                     <- other filters work

$ codegraph roles --role entry -T
No classified symbols found. Run "codegraph build" first.    <- misleading
```

`-j` is unaffected and correctly returns `{"count":0,…}`.

Reproduced on `ops-development-tool`, `data-analytics-pipeline-svc`, `tenant-optave-data-retrieval-svc`, `client-data-retrieval-scaffold-lib`, and `shared-infrastructure-and-data-lib` (the last also with `adapter`, `dead-entry`, and `dead-ffi`).

## Suggested fix

Distinguish the two cases — when a `--role` filter is active and the graph is non-empty, say so:

```
No symbols with role "entry". (548 classified symbols in graph.)
```

Only recommend `codegraph build` when the graph genuinely has no classified symbols.

Worth auditing the other `count === 0` branches in `queries-cli/` for the same conflation.

### PR DESCRIPTION
**Pull request by carlos-alm**
Closes #2390

## Problem

`codegraph roles --role <X>` with zero matches always said `No classified symbols found. Run "codegraph build" first.` — even on a fully and correctly built graph where the role simply has no members. Per the issue, this cost real time across 5 of 14 agents in an org rollout, each re-running `codegraph build` for nothing.

`rolesData`'s `count` conflates two different states:
- the graph has no classified symbols at all (genuinely worth rebuilding), and
- this `--role` filter matched nothing (the graph is fine).

## Fix

`rolesData` (`src/domain/analysis/roles.ts`) now also returns `totalClassified`: the classified-symbol count ignoring the `--role` filter but still respecting the `--file`/`-T` (noTests) filters. It's computed via one extra lightweight query, only when a role filter is active (free — identical to `rows.length` — in the common unfiltered case).

The CLI (`src/presentation/queries-cli/overview.ts`) uses this to print:

```
No symbols with role "entry". (548 classified symbols in graph.)
```

instead of the misleading rebuild suggestion, whenever the graph is otherwise non-empty. The original message is preserved for the genuine zero-graph case.

No native/Rust changes — this is a TypeScript query/presentation-layer fix only; `domain/analysis/` and `presentation/queries-cli/` have no Rust mirror (the dual-engine parity requirement applies to graph extraction, not CLI query formatting).

## Scope note

While auditing `queries-cli/` for the same conflation (per the issue's suggestion), found an identical bug in `codegraph exports <file>` (a file with legitimately zero exports gets told to rebuild). Filed separately as #2530 to keep this PR to one concern.

## Test plan

- [x] `tests/integration/roles.test.ts`: 3 new tests on `rolesData` — `totalClassified` matches the unfiltered count when a role filter has zero matches, equals `count` when no filter is applied, and respects the `file` filter.
- [x] `tests/presentation/queries-cli.test.ts`: 2 new tests on the CLI's `roles()` formatter — role-specific zero-match message when the graph is otherwise classified, and the original rebuild message preserved when the graph genuinely has none.
- [x] Verified all 5 new tests fail without the fix (reverted the two source files, confirmed failures, restored).
- [x] Full suite: `npm test` — 5313 passed.
- [x] `npm run lint` clean.
- [x] `codegraph diff-impact --staged -T` — bounded impact (2 files, 2 transitive callers, benchmark script + MCP passthrough only, both benign additive changes).

### CODE REVIEW

**Comment by carlos-alm** [comment]
@greptileai please re-review — added totalClassifiedUnscoped to handle the file/no-tests scoped-zero case you flagged, so the CLI no longer recommends a rebuild when a --file/--no-tests scope excludes everything but the graph is healthy elsewhere.

File: src/presentation/queries-cli/overview.ts [hunk 1]
```diff
@@ -369,7 +370,14 @@ export function roles(customDbPath: string, opts: OutputOpts = {}): void {
...
 
   if (data.count === 0) {
-    console.log('No classified symbols found. Run "codegraph build" first.');
+    if (opts.role && data.totalClassified > 0) {
+      // The graph is fine — this role filter simply matched nothing (#2390).
+      console.log(
+        `No symbols with role "${opts.role}". (${data.totalClassified} classified symbols in graph.)`,
+      );
+    } else {
+      console.log('No classified symbols found. Run "codegraph build" first.');
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply src/presentation/queries-cli/overview.ts:398]: This describes the exact scenario the previous commit (e5de8c27) already fixed — totalClassifiedUnscoped covers the file/no-tests-scoped-zero case, and it's covered by 2 new tests in that commit (both verified to fail without the fix via revert-and-rerun). This looks like a re-surfaced round-1 finding rather than a new one against e5de8c27; not making further changes for it. The genuinely new finding in this round (scoped count mislabeled as graph-wide) is fixed in d6afc742.

**Comment by carlos-alm** [comment]
@greptileai please re-review — fixed the scoped-count-mislabeled-as-graph-wide finding in d6afc742. Also replied inline on the other open comment: that one describes a scenario already fixed by the prior commit (e5de8c27), verified by 2 tests that fail without the fix — its diff_hunk shows it's anchored to the pre-e5de8c27 commit (a30efcd0), so it looks like a re-surfaced round-1 finding rather than a new one.

File: src/presentation/queries-cli/overview.ts [hunk 2]
```diff
@@ -369,7 +371,24 @@ export function roles(customDbPath: string, opts: OutputOpts = {}): void {
   if (outputResult(data, 'symbols', opts, customDbPath)) return;
 
   if (data.count === 0) {
-    console.log('No classified symbols found. Run "codegraph build" first.');
+    if (opts.role && data.totalClassified > 0) {
+      // The graph is fine — this role filter simply matched nothing (#2390).
+      console.log(
+        `No symbols with role "${opts.role}". (${data.totalClassified} classified symbols in graph.)`,
+      );
+    } else if (data.totalClassifiedUnscoped) {
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply src/presentation/queries-cli/overview.ts]: Fixed in d6afc742 — the role-specific branch now labels the count by its actual scope (file/no-tests) instead of always saying "in graph". Covered by 2 new tests (verified to fail without the fix via revert-and-rerun).
